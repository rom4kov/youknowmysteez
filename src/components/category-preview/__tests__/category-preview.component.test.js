import { screen } from "@testing-library/react";
import { renderWithProviders } from "../../../utils/test/test.utils";
import CategoryPreview from "../category-preview.component";

describe("CategoryPreview tests", () => {
  const title = "sneakers";
  const products = [
    {
      id: 1,
      name: "Black Converse",
      brand: "Converse",
      price: 100,
      imageUrl: "converse.com",
    },
    {
      id: 2,
      name: "Red Converse",
      brand: "Converse",
      price: 90,
      imageUrl: "converse.com",
    },
  ];

  test("should render CategoryPreview component correctly", () => {
    const { asFragment } = renderWithProviders(
      <CategoryPreview title={title} products={products} />,
      {
        preloadedState: {},
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("all products of products array are rendered", () => {
    renderWithProviders(<CategoryPreview title={title} products={products} />, {
      preloadedState: {},
    });

    const product1 = screen.getByText(/black converse/i);
    const product2 = screen.getByText(/red converse/i);
    expect(product1).toBeInTheDocument();
    expect(product2).toBeInTheDocument();
  });
});
