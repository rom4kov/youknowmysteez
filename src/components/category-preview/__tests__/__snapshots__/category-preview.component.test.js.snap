// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPreview tests should render CategoryPreview component correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-fsQiph dAVzcd"
  >
    <h2>
      <a
        class="sc-qRumB fuiFcU title"
        href="/sneakers"
      >
        SNEAKERS
      </a>
    </h2>
    <div
      class="sc-jsMahE juryKw"
    >
      <div
        class="sc-dmqHEX kneRcb"
      >
        <img
          alt="Black Converse"
          class="sc-hLseeU iwsRzB"
          src="converse.com"
        />
        <div
          class="sc-eDDNvR iqCncc"
        >
          <span
            class="sc-jTrPJq ckcNaj"
          >
            Black Converse
          </span>
          <span
            class="sc-gLDzan ewZTjn"
          >
            € 100
          </span>
        </div>
        <button
          class="buttonstyles__BaseButton-sc-lswy1h-0 buttonstyles__InvertedButton-sc-lswy1h-2 jJMbCT fsMtkj sc-iAEyYk jvcVwh"
        >
          In den Warenkorb
        </button>
      </div>
      <div
        class="sc-dmqHEX kneRcb"
      >
        <img
          alt="Red Converse"
          class="sc-hLseeU iwsRzB"
          src="converse.com"
        />
        <div
          class="sc-eDDNvR iqCncc"
        >
          <span
            class="sc-jTrPJq ckcNaj"
          >
            Red Converse
          </span>
          <span
            class="sc-gLDzan ewZTjn"
          >
            € 90
          </span>
        </div>
        <button
          class="buttonstyles__BaseButton-sc-lswy1h-0 buttonstyles__InvertedButton-sc-lswy1h-2 jJMbCT fsMtkj sc-iAEyYk jvcVwh"
        >
          In den Warenkorb
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
